modules/pel/peltool/user_data.py

```diff
@@ -0,0 +1,63 @@
+from pel.datastream import DataStream
+from collections import OrderedDict
+from pel_values import *
+from enum import Enum, unique
+import json
+
+
+@unique
+class UserDataFormat(Enum):
+    json = 0x1
+    cbor = 0x2
+    text = 0x3
+    custom = 0x4
+
+
+class UserData:
+    """
+    This represents the User Data section in a PEL.  It is free form data
+    that the creator knows the contents of.  The component ID, version,
+    and sub-type fields in the section header are used to identify the
+    format.
+
+    The Section base class handles the section header structure that every
+    PEL section has at offset zero.
+    """
+
+    def __init__(self, stream: DataStream, sectionID: int, sectionLen: int,
+                 versionID: int, subType: int, componentID: int, creatorID: str):
+        self.stream = stream
+        self.sectionID = sectionID
+        self.sectionLen = sectionLen
+        self.versionID = versionID
+        self.subType = subType
+        self.componentID = componentID
+        self.creatorID = creatorID
+        self.dataLength = sectionLen - 8
+        self.data = self.stream.get_mem(self.dataLength)
+
+    def paser(self) -> str:
+        import importlib
+        name = (self.creatorID.lower() + "%04X" % self.componentID).lower()
+        try:
+            cls = importlib.import_module("udparsers." + name + "." + name)
+            return cls.parseUDToJson(self.subType, self.versionID, self.data)
+        except:
+            print("Failed to parse user data, module = " + name)
```

Comment:
may as well also print error here, like
except Exception as e:
   print(... str(e))

Reply:
the import_module will throw if there isn't a module, and we don't expect modules for all PELs since PowerVM doesn't provide them.  So if import_module fails, the section can just be hex dumped.